src/Core/Error.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @param {Element} element
+ * @param {string} from
+ */
+function errorException (element, from) {
+	if (!(this instanceof Error))
+		return errorException.call(new Error(this), element, from)
+
+	var children = ''
+	var tabs = ''
+	var host = element
+	var stack = this.stack
+
+	while (host.type) {
+		children += tabs + '<' + getDisplayName(host.type) + '>\n'
+		tabs += '  '
+		host = host.host
+	}
+
+	this.children = children
+	this.from = from
+
+	console.error('Error caught in `\n\n'+children+'\n` from "'+from+'"\n\n'+stack+'\n\n')
```

Comment:
I'm still in favor of conditionally logging this in `errorRecovery` only if `LifecycleDidCatch` is unset
Just put the string on the error `this.dioStack`
https://github.com/thysultan/dio.js/issues/22

Reply:
What do you think are the downsides vs benefits of logging and not-logging?

Reply:
If I write `componentDidCatch`, it means I (might) know how to recover completely and gracefully (perhaps a `warn` or `debug` log).
In that case, logging the error would mistakenly alert my Ops team (esp. server side) that something needs immediate attention.

Reply:
Re: server side; In what case would a `console.error(...)` alert the ops team and why would that be a problem.

>If I write componentDidCatch, it means I (might) know how to recover completely and gracefully (perhaps a warn or debug log).

The current developer that is not necessarily the one that implemented `componentDidCatch` which makes it a benefit to log in this case since the author down stream needs to know that there is a component upstream that enters into an error state.

Reply:
Log aggregation of stdout, with an alert filter on error. A bit extreme (for a web service at least), but not unimaginable.

Even without alerting, it would show up in my logs as an `error` when (if expected) should be a `warn` or perhaps not at all. I don't want to have to filter out error logs that aren't actually errors.

I don't understand the author argument; If I publish a component where downstream clients need to know about error states I will propagate that information manually (because they might also be able to gracefully handle my error states)

Reply:
If the author of the `componentDidCatch` upstream does not log the error the author down stream is left in the dark when an error does occur since implementing `componentDidCatch` halts the bubbling of `componentDidCatch` from moving up the tree of components in search of the closest implementation.

Reply:
Proper authorship of `componentDidCatch` is to handle an expected (specific) error, otherwise re-throw / log.

Reply:
Yes, this argues for the case when proper authorship is not exhibited, which is the worse case that a not-logging pattern would miss.